Fix for Issue with Local testing on systems with non en-us Culture

ahpooch opened the pull request:
### Pull Request (PR) description
Fix for Issue with local testing on systems with non en-US Culture.

### This Pull Request (PR) fixes the following issues
- Fixes #283

### Task list
- [x] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [ ] Resource documentation added/updated in README.md.
- [ ] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [ ] Comment-based help added/updated.
- [x] Localization strings added/updated in all localization files as appropriate.
- [ ] Examples appropriately added/updated.
- [x] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [x] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285)
<!-- Reviewable:end -->

johlju commented on tests/Unit/Classes/DnsRecordPtr.tests.ps1:
```diff
@@ -484,22 +484,21 @@ Describe 'Test bad inputs (both IPv4 and IPv6)' -Tag 'Test', 'DnsRecord', 'DnsRe
             }
 
-            { $malformattedIPv4State.Get() } | Should -Throw -ExpectedMessage ('*' + '"Cannot convert value "{0}" to type "System.Net.IPAddress". Error: "An invalid IP address was specified.""' -f $malformattedIPv4State.IpAddress)
+            { $malformattedIPv4State.Get() } | Should -Throw -ErrorId 'MethodArgumentConversionInvalidCastArgument'
```
This does not seem correct to me, it looks like the resource doesn't throw an error using the correct method (probably wrongly uses `throw`) so we get an generic error instead of the actual DSC resource localized error. 

Can we get a better error by using the DSC community stamdard:
https://github.com/dsccommunity/SqlServerDsc/blob/1b3ea84057ba7dcbed5eca13d5a4ecde81068126/source/Classes/020.SqlPermission.ps1#L366-L371

ahpooch commented:
Implemented usage of Assert-IPAddress [in accordance with proposal](https://github.com/dsccommunity/DnsServerDsc/pull/285#pullrequestreview-2991438459)

I hope I've implemented the proposed idea correctly. The validation using Assert-IPAddress is only performed in the Get() area since other calls utilize it as I understand it.

ahpooch commented:
*[`tests/Unit/Classes/DnsRecordPtr.Tests.ps1` line 161 at r4](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285#-OUihlzg7P2Mb3BH0AXU:-OUnJ-KP8DgeXm5OqjI8:b-uo408m) ([raw file](https://github.com/dsccommunity/DnsServerDsc/blob/c1e823acff92579dc7778e77e4a2b260aff34632/tests/Unit/Classes/DnsRecordPtr.Tests.ps1#L161)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Same here
</blockquote></details>

Answered to a previous message.


<!-- Sent from Reviewable.io -->

ahpooch commented:
*[`tests/Unit/Classes/DnsRecordMxScoped.tests.ps1` line  at r4](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285#-OUiibQB0heE3B-IT5UT:-OUnJ9dp08jGOI8nhGvw:be29usg) ([raw file](https://github.com/dsccommunity/DnsServerDsc/blob/c1e823acff92579dc7778e77e4a2b260aff34632/tests/Unit/Classes/DnsRecordMxScoped.tests.ps1#L)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

The filename was renamed here from `Tests.ps1` to `tests.ps1`, was there a reason?
</blockquote></details>

While working on the project earlier, I noticed inconsistency in the filenames `*.tests.ps1` and `*.Tests.ps1`. In previous PRs, I had already renamed files to follow the `*.Tests.ps1` standard when working on them. In this commit, I likely got confused between the state of the working directory files and the Git index.

I added another commit to fix this.


<!-- Sent from Reviewable.io -->

ahpooch reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285)* status: 5 of 6 files reviewed, 2 unresolved discussions (waiting on @johlju)

___
*[`tests/Unit/Classes/DnsRecordPtr.Tests.ps1` line 108 at r4](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285#-OUihYvM7NajPGh7Ovxq:-OUtvOmg2umoVhWL1h3Z:b-896fix) ([raw file](https://github.com/dsccommunity/DnsServerDsc/blob/c1e823acff92579dc7778e77e4a2b260aff34632/tests/Unit/Classes/DnsRecordPtr.Tests.ps1#L108)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

It was probably debug messages during development that was missed being removed prior to merge. Suggest fixing it in this PR now and then create an issue to resolve the rest. 🙂
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

johlju reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/dsccommunity/DnsServerDsc/285#-:-OUyqT0qCbj3aoHlZTjd:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @ahpooch)



<!-- Sent from Reviewable.io -->
